BaikalDataGenerator/Source/utils.h

```diff
@@ -37,9 +38,52 @@ struct DGenConfig
     size_t width, height;
     size_t split_num = 1;
     size_t split_idx = 0;
-    size_t offset_idx = 0;
+    std::int32_t offset_idx = 0;
     std::uint32_t num_bounces = 5;
     bool gamma_correction;
 };
 
+template<typename T>
+std::vector<std::vector<T>> SplitVector(const std::vector<T>& vec, size_t n)
+{
+    std::vector<std::vector<T>> splits;
+
+    size_t length = vec.size() / n;
+    size_t remain = vec.size() % n;
+
+    size_t begin = 0;
+    size_t end = 0;
+
+    for (size_t i = 0; i < std::min(n, vec.size()); ++i)
+    {
+        end += (remain > 0) ? (length + ((remain--) != 0)) : length;
+        splits.push_back(std::vector<T>(vec.begin() + begin, vec.begin() + end));
+        begin = end;
+    }
+
+    return splits;
+}
+
+template<typename T>
+std::vector<T> GetSplitByIdx(const std::vector<T>& vec, size_t n, size_t idx)
```

Comment:
There can be e.g. `std::list`. So I guess the function might take `const Range& range` and return something like `std::pair<typename Range::const_iterator, typename Range::const_iterator>`.